File: javascript/6_kyu/duplicate_encoder.js
// Duplicate Encoder
// https://www.codewars.com/kata/54b42f9314d9229fd6000d9c/train/javascript

function duplicateEncode(word) {
  word = word.toLowerCase();
  var charCount = {};
  var ans = [];

  for (var i = 0; i < word.length; i++) {
    if ( word[i] in charCount ) {
      charCount[word[i]]++;
    } else {
      charCount[word[i]] = 1;
    }
  }

  for (var key = 0; key < word.length; key++) {
    if (charCount[word[key]] == 1) {
      ans.push("(");
    } else {
      ans.push(")");
    }
  }

  return ans.join("");
}


console.log(duplicateEncode("recede"))

File: python/6_kyu/duplicate_encoder.py
# Duplicate Encoder
# https://www.codewars.com/kata/54b42f9314d9229fd6000d9c/train/python

# Solution 1
def duplicate_encode(word):
  word = word.lower()
  ans = []

  for i in word:
    if word.count(i) > 1:
      ans.append(")")
    else:
      ans.append("(")

  return ''.join(ans)

# Solution 2: More Pythonic
def duplicate_encode(word):
  word = word.lower()
  return ''.join([")" if word.count(i) > 1 else "(" for i in word])

print(duplicate_encode("Success"))